### ISSUE TITLE
Build error with filenames containing unicode characters

### ISSUE THREAD

**Issue by bpabel** [takluyver/flit#345]
I'm trying to build a project that has files that contain unicode characters in their filename.  I can build the wheel successfully, but building the sdist results in the following error.

<img width="351" alt="Screen Shot 2020-05-20 at 11 34 07 PM" src="https://user-images.githubusercontent.com/622914/82530788-ee5c2200-9af2-11ea-9f33-9998f6422a59.png">

```
$ flit build
Found 6 files tracked in git                                                                                                                                   I-flit.sdist
Traceback (most recent call last):
  File "/Users/brendan/.local/bin/flit", line 8, in <module>
    sys.exit(main())
  File "/Users/brendan/.local/pipx/venvs/flit/lib/python3.8/site-packages/flit/__init__.py", line 170, in main
    main(args.ini_file, formats=set(args.format or []),
  File "/Users/brendan/.local/pipx/venvs/flit/lib/python3.8/site-packages/flit/build.py", line 46, in main
    sdist_file = sb.build(dist_dir, gen_setup_py=gen_setup_py)
  File "/Users/brendan/.local/pipx/venvs/flit/lib/python3.8/site-packages/flit/sdist.py", line 223, in build
    return Path(super().build(str(target_dir), gen_setup_py=gen_setup_py))
  File "/Users/brendan/.local/pipx/venvs/flit/lib/python3.8/site-packages/flit_core/sdist.py", line 185, in build
    ti = tf.gettarinfo(path, arcname=pjoin(self.dir_name, relpath))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/tarfile.py", line 1830, in gettarinfo
    statres = os.lstat(name)
FileNotFoundError: [Errno 2] No such file or directory: '"tests/No\\303\\253l.txt"'
```

**Comment by takluyver** [comment]
Thanks. It looks like `git ls-files` quotes the filename for us when it's called here:

https://github.com/takluyver/flit/blob/cfd71d81be95300266c6e989cf2d1bf57aa637fc/flit/vcs/git.py#L6-L9

I see it has a `-z` option to give the raw filenames delimited by null bytes. That's probably what we need.

**Comment by bpabel** [comment]
@takluyver You were exactly right, thanks!  Changing that fixed the issue (see #346 )
